alf/experience_replayers/replay_buffer.py

```diff
@@ -99,18 +144,72 @@ def _create_buffer(tensor_spec):
             self._flattened_buffer = alf.nest.map_structure(
                 lambda x: bs.flatten(x), self._buffer)
 
+        if allow_multiprocess:
+            self.share_memory()
+
     def check_convert_env_ids(self, env_ids):
         with alf.device(self._device):
             if env_ids is None:
                 env_ids = torch.arange(self._num_envs)
             else:
                 env_ids = env_ids.to(torch.int64)
             env_ids = _convert_device(env_ids)
+            assert len(env_ids.shape) == 1, "env_ids should be a 1D tensor"
             return env_ids
 
-    def enqueue(self, batch, env_ids=None):
+    def has_space(self, env_ids):
+        """Check free space for one batch of data for env_ids.
+
+        Args:
+            env_ids (Tensor): Assumed not None, properly checked by
+                check_convert_env_ids().
+        
+        Returns:
+            bool
+        """
+        current_size = self._current_size[env_ids]
+        max_size = current_size.max()
+        return max_size < self._max_length
+
+    def enqueue(self, batch, env_ids=None, blocking=False):
         """Add a batch of items to the buffer.
 
+        Args:
```

Comment:
need to document the be behavior of block=False: it  always success even if there is no free slot. If there is no free slot, ...

The typical for block=False is that if there is not free space, it will not do anything and return False.